test/Microsoft.AspNet.Mvc.FunctionalTests/HtmlGenerationTest.cs

```diff
@@ -575,5 +575,22 @@ public class HtmlGenerationTest
             // Assert
             Assert.Equal(expected, response);
         }
+
+        [Fact]
+        public async Task AttributeWithBooleanValues_RendersWithCorrectSpecialCasing()
+        {
+            // Arrange
+            var server = TestHelper.CreateServer(_app, SiteName, _configureServices);
+            var client = server.CreateClient();
+            var outputFile = "compiler/resources/HtmlGenerationWebSite.HtmlGeneration_Home.AttributesWithBooleanValues.html";
+            var expectedContent =
+                await ResourceFile.ReadResourceAsync(_resourcesAssembly, outputFile, sourceFile: false);
+
+            // Act
+            var response = await client.GetStringAsync("http://localhost/HtmlGeneration_Home/AttributesWithBooleanValues");
+
+            // Assert
+            Assert.Equal(expectedContent, response);
+        }
     }
```

Comment:
Should add a test variation that encodes the same file with the `TestHtmlEncoder`. Will make sure the new `WriteAttributeValue` method encodes correctly in the new cases.